bot/pagination.py

```diff
@@ -87,6 +108,53 @@ def add_line(self, line: str = '', *, empty: bool = False) -> None:
             self._current_page.append('')
             self._count += 1
 
+        if remaining_words:
+            self._new_page()
+            self.add_line(remaining_words)
+
+    def _new_page(self) -> None:
+        self._linecount = 0
+        self._count = len(self.prefix) + 1
+        self.close_page()
+
+    def _split_remaining_words(self, line: str, max_chars: int) -> t.Tuple[str, t.Optional[str]]:
+        """
+        Internal: split a line into two strings -- reduced_words and remaining_words.
+
+        reduced_words: the remaining words in `line`, after attempting to remove all words that
+            exceed `max_chars` (rounding down to the nearest word boundary).
+
+        remaining_words: the words in `line` which exceed `max_chars`. This value is None if
+            no words could be split from `line`.
+
+        Return a tuple in the format (reduced_words, remaining_words).
+        """
+        reduced_words = []
+        remaining_words = []
+        # "(Continued)" is used on a line by itself to indicate the continuation of last page
```

Comment:
Nitpick, but block comments trapped between two lines of code feel really claustrophobic to me.
```suggestion

        # "(Continued)" is used on a line by itself to indicate the continuation of last page
```